compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1167,15 +1168,35 @@ object SymDenotations {
       privateWithin: Symbol = null,
       annotations: List[Annotation] = null)(implicit ctx: Context) =
     { // simulate default parameters, while also passing implicit context ctx to the default values
-      val initFlags1 = (if (initFlags != UndefinedFlags) initFlags else this.flags) &~ Frozen
+      val initFlags1 = (if (initFlags != UndefinedFlags) initFlags else this.flags)
       val info1 = if (info != null) info else this.info
+      if (ctx.isAfterTyper && changedClassParents(info, info1, completersMatter = false))
+        assert(ctx.phase.changesParents, i"undeclared parent change at ${ctx.phase} for $this, was: $info, now: $info1")
```

Comment:
The check is in `enteredAfter`. I think it would make little sense checking in `info_=` because typically we set the info with a clone of the previous scope and then add symbols later using `enteredAfter`.